packages/fiori/src/upload-utils/BodyDragAndDrop.js

```diff
@@ -0,0 +1,79 @@
+
+/**
+ * Handles drag and drop event listeners on document.body.
+ * Ensures that there is only 1 listener per type attached (drag, drop, leave). Event listeners will only be attached when
+ * there is at least 1 UploadCollection registered in the set.
+ */
+
+import UploadCollectionDnDOverlayMode from "../types/UploadCollectionDnDMode.js";
+
+const draggingFiles = event => {
+	return event.dataTransfer.types.includes("Files");
+};
+
+const uploadCollections = new Set();
+let lastDragEnter = null;
+let globalHandlersAttached = false;
+
+const ondragenter = event => {
+	if (!draggingFiles(event)) {
+		return;
+	}
+
+	lastDragEnter = event.target;
+	uploadCollections.forEach(uc => {
+		if (uc._dndOverlayMode !== UploadCollectionDnDOverlayMode.Drop) {
```

Comment:
The right pattern for this kind of behavior is: publish/subscribe. This file should not touch the private APIs of upload collections. They should register events and have this file fire the events, and they should do the work themselves.
See EventProvider.js in the base project. Use that as in OpenUI5

As a result, the methods this will export will be called something like attachDnDHandler, detachDnDHandler rather than add/remove upload collection instance

Reply:
Done